lib/ddtrace/sampling/rule_sampler.rb

```diff
@@ -97,18 +97,30 @@ def sample_span(span)
         sampled = rule.sample?(span)
         sample_rate = rule.sample_rate(span)
 
+        set_priority(span, sampled)
         set_rule_metrics(span, sample_rate)
 
         return false unless sampled
 
-        rate_limiter.allow?(1).tap do
+        rate_limiter.allow?(1).tap do |allowed|
+          set_priority(span, allowed)
           set_limiter_metrics(span, rate_limiter.effective_rate)
         end
       rescue StandardError => e
         Datadog.logger.error("Rule sampling failed. Cause: #{e.message} Source: #{Array(e.backtrace).first}")
         yield(span)
       end
 
+      # Span priority should only be set when the {RuleSampler}
+      # was responsible for the sampling decision.
+      def set_priority(span, sampled)
+        if sampled
+          ForcedTracing.keep(span)
```

Comment:
Better not to use `ForcedTracing` as it was deprecated.

I would assign directly to the `span.context` itself, because we're trying to remove this "force sampling by span tag" behavior in 1.0. Either way, its going to cause a merge conflict in 1.0, but replacing the "context sampling assignment" has a more direct analog in 1.0.

Reply:
Changed, thank you!

Reply:
I ended up reverting it back to `ForcedTracing`, as it handles cases where `span.context` is nil, and I'd have to duplicate that here.
I believe all this will change in 1.0, so I'm not too concerned at this moment.